### ISSUE TITLE
mbtiles plugin stopped working between JOSM 9605 and 9624

### ISSUE THREAD

**Issue by PaulUithol** [iandees/josm-mbtiles#8]
Sorry to be filing a very similar bug to last time, but the upgrade from JOSM 9329 to 9979 seems to break rendering of mbtiles again..

On a clean JOSM with only mbtiles installed, starting up and loading an mbtiles file yields the following messages:

```
C:\Program Files (x86)\JOSM>java -d64 -Xmx4096m -jar josm-tested.jar
INFO: loading plugin 'mbtiles' (version v2.0)
INFO: Open file: C:\Users\Paul\Desktop\Kampala_North.mbtiles (245753856 bytes)
[INFO] CompositeCacheManager - Instance is null, returning unconfigured instance
[INFO] ThreadPoolManager - thread_pool.default PoolConfiguration = useBoundary = [false] boundarySize = [2000] maximumPoolSize = [150] minimumPoolSize = [4] keepAliveTime = [300000] whenBlockedPolicy = [RUN] startUpSize = [4]
[INFO] CompositeCacheManager - Setting default auxiliaries to null
[INFO] CompositeCacheManager - setting defaultCompositeCacheAttributes to [ useLateral = true, useRemote = true, useDisk = true, maxObjs = 1000, maxSpoolPerRun = -1, diskUsagePattern = UPDATE, spoolChunkSize = 2 ]
[INFO] CompositeCacheManager - setting defaultElementAttributes to [ IS_LATERAL = true, IS_SPOOL = true, IS_REMOTE = true, IS_ETERNAL = false, MaxLifeSeconds = -1, IdleTime = -1, CreateTime = 1458821614572, LastAccessTime = 1458821614572, getTimeToLiveSeconds() = -1, createTime = 1458821614572 ]
[INFO] CompositeCacheConfigurator - Parsed regions []
[INFO] CompositeCacheManager - Finished configuration in 31 ms.
[INFO] CompositeCacheConfigurator - No special ElementAttribute class defined for key [jcs.region.TMS.elementattributes], using default class.
[INFO] AbstractDoubleLinkedListMemoryCache - initialized MemoryCache for TMS
[INFO] CompositeCache - Constructed cache with name [TMS] and cache attributes [ useLateral = true, useRemote = true, useDisk = true, maxObjs = 0, maxSpoolPerRun = -1, diskUsagePattern = UPDATE, spoolChunkSize = 2 ]
[INFO] IndexedDiskCache - Region [TMS] Cache file root directory: C:\Users\Paul\AppData\Local\JOSM\cache\tiles
[INFO] IndexedDiskCache - Region [TMS] Set maxKeySize to: '524288'
[INFO] IndexedDiskCache - Region [TMS] Loaded keys from [TMS], key count: 5300; up to 524288 will be available.
[INFO] IndexedDiskCache - Region [TMS] Finished inital consistency check, isOk = true in 0ms.
[INFO] IndexedDiskCache - Region [TMS] Indexed Disk Cache is alive.
```

However, nothing shows up in the layers panel or on the canvas in 9979. In JOSM 9329, it still works fine.

**Comment by PaulUithol** [comment]
Tracking back, looks like this issue is present in 9900 already

**Comment by iandees** [comment]
Thanks for filing this. I'm on vacation and will take a look next week when
I get back.

**Comment by PaulUithol** [comment]
Thank you very much. Have pinpointed the breakage to be between these two versions:

josm-snapshot-9605.jar	2016-01-24 03:34 	10M	 
josm-snapshot-9624.jar	2016-01-25 03:34 	10M	 

Still works on 9605, doesn't anymore on 9624. Seems to be something in the later stage of loading the file; the messages look the same, but no layer & no entry will get added to the "Open Recent" menu either.

**Comment by PaulUithol** [comment]
https://josm.openstreetmap.de/log/josm/?action=stop_on_copy&mode=stop_on_copy&rev=9624&stop_rev=9605&limit=100&verbose=on&sfp_email=&sfph_mail=

9620 reads "Refactor structure of TMS based classes. Move OsmMercator initialization to TMSTileSource and switch all TMS based TileSource so they extend TMSTileSource. Now WMTSTileSource will not create OsmMercator object that it will not use anyhow."

**Comment by don-vip** [comment]
Probably fixed in #9
